src/dvc_data/index/index.py

```diff
@@ -710,21 +711,17 @@ def traverse(self, *args, **kwargs) -> Any:
     def iteritems(
         self,
         prefix: Optional[DataIndexKey] = None,
-        shallow: Optional[bool] = False,
+        shallow: bool = False,
     ) -> Iterator[Tuple[DataIndexKey, DataIndexEntry]]:
-        kwargs: Dict[str, Any] = {"shallow": shallow}
         if prefix:
-            kwargs = {"prefix": prefix}
             item = self._trie.longest_prefix(prefix)
             if item:
                 key, entry = item
                 self._load(key, entry)
 
-        # FIXME could filter by loaded and/or isdir in sql on sqltrie side
-        for key, entry in self._trie.items(**kwargs):
+        for key, entry in self._trie.items(prefix=prefix, shallow=shallow):
             self._load(key, entry)
-
-        yield from self._trie.items(**kwargs)
+            yield key, entry
```

Comment:
sqltrie `items()` allows for loading children during iteration now so we do not need to iterate over the trie twice